helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/WagedRebalancer.java

```diff
@@ -413,28 +432,57 @@ private void globalRebalance(ResourceControllerDataProvider clusterData,
             HelixRebalanceException.Type.INVALID_CLUSTER_STATUS, ex);
       }
 
-      calculateAndUpdateBaseline(clusterModel, algorithm);
+      final boolean waitForGlobalRebalance = !_asyncBaselineCalculation;
```

Comment:
Prefer to use a consistent boolean variable, _waitForGlobalRebalance is sufficient without `_asyncBaselineCalculation`

Reply:
_waitForGlobalRebalance at the rebalancer level does not make much sense.
And we need to pass it into 2 methods with different values. To ensure these 2 methods get the same boolean value, we need to assign it to some final local variable.

The alternative way is localAsyncBoolean = _asyncBoolean;
But I feel changing the name a little bit would help itself explain.